consultation_analyser/consultations/views/consultations.py

```diff
@@ -29,14 +30,27 @@ def index(request: HttpRequest) -> HttpResponse:
 
 @user_can_see_consultation
 @login_required
-def show(request: HttpRequest, consultation_slug: str) -> HttpResponse:
+def show(
+    request: HttpRequest, consultation_slug: str, processing_run_slug: Optional[str] = None
+) -> HttpResponse:
     consultation = get_object_or_404(models.Consultation, slug=consultation_slug)
-    questions = models.Question.objects.filter(section__consultation__slug=consultation_slug)
-    context = {"questions": questions, "consultation": consultation}
-
-    if not consultation.has_processing_run():
+    if processing_run_slug:
```

Comment:
this is duplicated in three places — worth turning into a method on `Consultation` that accepts an optional `processing_run_slug`?